.github/actions/install-aiida-core/action.yml

```diff
@@ -0,0 +1,50 @@
+name: Install aiida-core
+description: Install aiida-core package and its Python dependencies
+
+inputs:
+  python-version:
+    description: Python version
+    default: '3.9'              # Lowest supported version
+    required: false
+  extras:
+    description: aiida-core extras (including brackets)
+    default: ''
+    required: false
+  # NOTE: Hard-learned lesson: we cannot use type=boolean here, apparently :-(
+  # https://stackoverflow.com/a/76294014
+  from-requirements:
+    description: Install aiida-core dependencies from pre-compiled requirements.txt file
+    default: 'true'
+    required: false
+
+runs:
+  using: composite
+  steps:
+  - name: Set Up Python
+    uses: actions/setup-python@v5
+    with:
+      python-version: ${{ inputs.python-version }}
+
+  - name: Install uv installer
+    run: curl --proto '=https' --tlsv1.2 -LsSf https://${{ env.UV_URL }} | sh
+    env:
+      UV_VERSION: 0.1.35
+      UV_URL: github.com/astral-sh/uv/releases/download/$UV_VERSION/uv-installer.sh
+    shell: bash
+
+  - name: Install dependencies from requirements-py-*.txt
+    if: ${{ inputs.from-requirements == 'true' }}
+    run: uv pip install --system -r requirements/requirements-py-${{ inputs.python-version }}.txt
+    shell: bash
+
+  - name: Install aiida-core
+    run: uv pip install --system ${{ env.NO_DEPS }} -e .${{ inputs.extras }}
+    env:
+        # Don't install dependencies if they were installed through requirements file AND
+        # if no extras are required.
+        #
+        # If this syntax looks weird to you, dear reader, know that this is
+        # GHA's way to do ternary operator. :-/
+        # https://docs.github.com/en/actions/learn-github-actions/expressions#example
```

Comment:
Just out of curiosity, what part do you find weird? To me the structure `(condition) && true || false` seems quite logical coming from bash. Then there is the `${{ }}` as GHA's outer expression wrapper.